### ISSUE TITLE
npm install fails through http proxy

### ISSUE THREAD

**Issue by andreas689** [giggio/node-chromedriver#255]
Unfortunately, it is not possible to install chromedriver via an http proxy. Proxy values are set as follows and the values are available in .npmrc:

```
npm config set proxy http://[user:pwd]@domain.tld:port
npm config set https-proxy http://[user:pwd]@domain.tld:port

```
If I try to install chromedriver via cli (npm install chromedriver --save-dev) I get the following error:
```
ChromeDriver installation failed Error: Error with http(s) request: Error: connect ETIMEDOUT 172.217.20.16:443
    at requestBinary (c:\dev\wdio-debugging\node_modules\chromedriver\install.js:254:11)


```
We have already done debugging and the problem we found is, installer fetches the https URL through the proxy by means of an http GET instead of an http CONNECT as you can see below(tcpdump).  
```
GET https://chromedriver.storage.googleapis.com/80.0.3987.16/chromedriver_win32.zip HTTP/1.1
Accept: application/json, text/plain, */*
User-Agent: npm/6.13.4 node/v12.16.1 win32 x64
host: chromedriver.storage.googleapis.com
Connection: close
```

If I change the custom binaries URL to http(npm install chromedriver --save-dev --chromedriver_cdnurl=http://chromedriver.storage.googleapis.com), the installation works. 
Could you please check?

**Comment by giggio** [comment]
This seems related to #252 which in turn is related to a bug in Axios: https://github.com/axios/axios/issues/2072

There are some workarounds suggested on the issue, if you could try them and send a PR, I'd appreciate. I don't have a proxy I can use to test right now.

**Comment by vasu512** [comment]
Any status on this. Axios:  [axios/axios#2072](https://github.com/axios/axios/issues/2072) seems to be closed

**Comment by giggio** [comment]
Yes, the status is: I don't have a proxy to test and we need someone who does to submit a PR.

**Comment by andreas689** [comment]
I will try workaround with https-proxy-agent and if the workaround is successful, I will send PR.

**Comment by vasu512** [comment]
Is this issue closed now? We are still waiting for the fix. There is no workaround too and this a very important module in our project.

**Comment by giggio** [comment]
I'm waiting for a PR. If this issue is so important I'm sure you can spend some minutes on it and send a PR.

**Comment by andreas689** [comment]
> 
> 
> Is this issue closed now? We are still waiting for the fix. There is no workaround too and this a very important module in our project.

I have already provided a sort a workaround above. You can use option _Custom binaries url_ --chromedriver_cdnurl  and download binary whit http and not with https.

**Comment by juangavaldes** [comment]
@giggio Do you guys need help testing this? I can use our proxy to test it, if you can give me some instructions

**Comment by giggio** [comment]
@juangavaldes yes! The instructions to use the proxy are here: https://github.com/giggio/node-chromedriver#custom-download-options
It seems people are having problems when the proxy is http, and they use an https download url.

**Comment by andreas689** [comment]
Hi @giggio, not only via http-proxy, also via https-proxy. I have developed a workaround with https-proxy-agent. Could you please create a branch from master with name bugifx_npm_install_fails_through_http_proxy. I will commit my minor changes afterwards.
Regards,
Andreas

**Comment by svettwer** [comment]
Sorry for hopping into the discussion but I subscribed the issue a few days ago and just saw the notification of your komment @andreas689.

On GitHub it is not easily possible to push into the original (so called upstream) repository. The workflow would be to fork the repository to your own namespace, add your changes there and then open a PR. More information can be found here: https://guides.github.com/activities/forking/

If I just misunderstood your comment and you're allowed to push to the upstream repo, just ignore me. :smile: 

BR
Sven

**Comment by giggio** [comment]
Hey @svettwer, thanks for stepping in. @andreas689 Sven is correct, you can create a fork and then submit a pull request.

**Comment by giggio** [comment]
Closed by #275. Published now as `83.0.1`.
